### PR TITLE
initial implementation of SpeakerEncoder forward function

### PR DESCRIPTION
**Pull request by richieyoum**
Hey team, this is my initial implementation of the forward function for SpeakerEncoder.

I'm not too certain what the author is referring to in this part of the [paper](https://arxiv.org/pdf/1806.04558.pdf): `Input 40-channel log-mel spectrograms are passed to a network consisting of a stack of 3 LSTM layers of 768 cells, each followed by a projection to 256 dimensions.`.

My interpretation is that, out of the 3 lstm layers, the initial lstm expects an input size of 40 (40 channels = input size of 40? unsure about the input shape as it's not implemented yet) and output dim of 256, and the rest of layers are set to 256 for both input and output dims.

However, looking at @CorentinJ's [implementation](https://github.com/CorentinJ/Real-Time-Voice-Cloning/blob/0713f860a3dd41afb56e83cff84dbdf589d5e11a/encoder/model.py#L20), he used same sizes for all 3 of the layers by simply putting `num_layers` to 3.

What do you guys think?

### CODE REVIEW

**Comment by pieceOpiland** [comment]
~~Shouldn't the additional 256 dimensions get reintroduced through the hidden or cell states that get propagated forward? I think it's ok to go with a single 3 layer LSTM block.~~

Edit: Nevermind, the architecture makes sense, I think it's not much cause for concern per my second comment.

**Comment by pieceOpiland** [comment]
> Shouldn't the additional 256 dimensions get reintroduced through the hidden or cell states that get propagated forward? I think it's ok to go with a single 3 layer LSTM block.

At any rate, the [pytorch LSTM](https://pytorch.org/docs/stable/generated/torch.nn.LSTM.html) doc for `num_layers` seems to indicate that it should handle the case.
